src/inc/internal/BundleWriterHelper.hpp

```diff
@@ -0,0 +1,95 @@
+#pragma once
+
+#include "AppxPackaging.hpp"
+#include "ComHelper.hpp"
+#include "DirectoryObject.hpp"
+#include "AppxBlockMapWriter.hpp"
+#include "ContentTypeWriter.hpp"
+#include "ZipObjectWriter.hpp"
+#include "BundleManifestWriter.hpp"
+#include "AppxPackageInfo.hpp"
+
+#include <map>
+#include <memory>
+#include <future>
+
+namespace MSIX {
+
+    struct PackageInfo
+        {
+            APPX_BUNDLE_PAYLOAD_PACKAGE_TYPE type;
+            UINT64 version;
+            std::string architecture;
+            std::string resourceId;
+            PCWSTR fileName;
+            UINT64 size;
+            UINT64 offset;
+            //ComPtr<IAppxManifestQualifiedResourcesEnumerator> resources;
+            ComPtr<IAppxManifestResourcesEnumerator> resources;
+
+            BOOL isDefaultApplicablePackage;
+            ComPtr<IAppxManifestTargetDeviceFamiliesEnumerator> tdfs;
+        };
+
+    class BundleWriterHelper

```

Comment:
>BundleWriterHelper [](start = 10, length = 18)

This class looks a lot like windows code, instead of the general idea of cross-plat application. 

You can change the contract here to don't be hresult based and use exceptions. also that way you can avoid having out parameters and just return the value

Reply:
Updated most methods in this file to use exceptions instead of hresults.